apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts

```diff
@@ -0,0 +1,515 @@
+     * @returns {number}  Tds taxable Amount
+     * @memberof AdvanceReceiptAdjustmentComponent
+     */
+    public calculateTdsAmount(productAmount: number, rate: number): number {
+        let taxAmount: number = 0;
+        let amount: number = 0;
+        amount = cloneDeep(Number(productAmount));
+        taxAmount = Number((amount * rate) / 100);
+        return Number(taxAmount.toFixed(2));
+    }
+
+    /**
+     * To save advance receipt adjustment
+     *
+     * @memberof AdvanceReceiptAdjustmentComponent
+     */
+    public saveAdjustAdvanceReceipt(): void {
+        let isValid: boolean = true;
+        if (this.isTaxDeducted) {
+            if (this.adjustVoucherForm.tdsTaxUniqueName === '') {
+                if (this.tdsTypeBox && this.tdsTypeBox.nativeElement)
+                    this.tdsTypeBox.nativeElement.classList.add('error-box');
+                isValid = false;
+            } else if (this.adjustVoucherForm.tdsAmount === '') {
+                if (this.tdsAmountBox && this.tdsAmountBox.nativeElement) {
+                    this.tdsAmountBox.nativeElement.classList.add('error-box');
+                    isValid = false;
+                }
+            }
+        } else {
+            delete this.adjustVoucherForm['tdsAmount'];
+            delete this.adjustVoucherForm['description'];
+            delete this.adjustVoucherForm['tdsTaxUniqueName'];
+        }
+        if ((Number(this.adjustPayment.grandTotal) - Number(this.adjustPayment.totalAdjustedAmount)) < 0) {
+            this.toaster.errorToast('The adjusted amount of the linked invoice\'s is more than this receipt');
+            isValid = false;
+        }
+        if (this.adjustVoucherForm && this.adjustVoucherForm.adjustments && this.adjustVoucherForm.adjustments.length > 1) {
+            this.adjustVoucherForm.adjustments.forEach(item => {
+                if (!item.voucherNumber) {
+                    isValid = false;
+                } else if (!item.dueAmount.amountForAccount) {
+                    isValid = false;
+                }
+            });
+        }
+        if (isValid) {
+            this.submitClicked.emit({
+                adjustVoucherData: this.adjustVoucherForm,
+                adjustPaymentData: this.adjustPayment
+            });
+        }
+    }
+
+    /**
+     * Select voucher type
+     *
+     * @param {IOption} event
+     * @param {number} index
+     * @memberof AdvanceReceiptAdjustmentComponent
+     */
+    public selectVoucher(event: IOption, entry: Adjustment, index: number): void {
+        if (event && entry) {
+            entry = cloneDeep(event.additional);
+            this.adjustVoucherForm.adjustments.splice(index, 1, entry);
+            this.calculateTax(entry, index);
+        }
+    }
+
+    /**
+     *  To handle click event of selected voucher sh-select
+     *
+     * @memberof AdvanceReceiptAdjustmentComponent
+     */
+    public clickSelectVoucher(index: number, form: NgForm): void {
```

Comment:
we are using return in function so return type can't be void

Reply:
Done